volatility/framework/plugins/windows/dumpfiles.py

```diff
@@ -0,0 +1,245 @@
+
+        # The DataSectionObject and ImageSectionObject caches are handled in basically the same way.
+        # We carve these "pages" from the memory_layer.
+        for member_name, extension in [("DataSectionObject", "dat"), ("ImageSectionObject", "img")]:
+            try:
+                section_obj = getattr(file_obj.SectionObjectPointer, member_name)
+                control_area = section_obj.dereference().cast("_CONTROL_AREA")
+                if control_area.is_valid():
+                    dump_parameters.append((control_area, memory_layer, extension))
+            except exceptions.InvalidAddressException:
+                vollog.log(constants.LOGLEVEL_VVV,
+                           "{0} is unavailable for file {1:#x}".format(member_name, file_obj.vol.offset))
+
+        # The SharedCacheMap is handled differently than the caches above.
+        # We carve these "pages" from the primary_layer.
+        try:
+            scm_pointer = file_obj.SectionObjectPointer.SharedCacheMap
+            shared_cache_map = scm_pointer.dereference().cast("_SHARED_CACHE_MAP")
+            if shared_cache_map.is_valid():
+                dump_parameters.append((shared_cache_map, primary_layer, "vacb"))
+        except exceptions.InvalidAddressException:
+            vollog.log(constants.LOGLEVEL_VVV,
+                       "SharedCacheMap is unavailable for file {0:#x}".format(file_obj.vol.offset))
+
+        for memory_object, layer, extension in dump_parameters:
+            cache_name = EXTENSION_CACHE_MAP[extension]
+            desired_file_name = "file.{0:#x}.{1:#x}.{2}.{3}.{4}".format(file_obj.vol.offset,
+                                                                        memory_object.vol.offset,
+                                                                        cache_name,
+                                                                        ntpath.basename(obj_name),
+                                                                        extension)
+
+            result_text = self.dump_file_producer(file_obj, memory_object, layer, desired_file_name)
+
+            yield (cache_name, format_hints.Hex(file_obj.vol.offset),
+                ntpath.basename(obj_name), # temporary, so its easier to visualize output
+                result_text)
+
+    def _generator(self, procs: List, offsets: List):
+
+        if procs:
+            # The handles plugin doesn't expose any staticmethod/classmethod, and it also requires stashing
+            # private variables, so we need an instance (for now, anyway). We _could_ call Handles._generator()
+            # to do some of the other work that is duplicated here, but then we'd need to parse the TreeGrid
+            # results instead of just dealing with them as direct objects here.
+            handles_plugin = handles.Handles(context=self.context, config_path=self._config_path)
+            type_map = handles_plugin.get_type_map(context=self.context,
```

Comment:
I'm surprised these aren't classmethods?  Definitely looks like one to fix up...